### PR TITLE
Fix test that doesn't compile when the JDK is Java 11 and remove a few tests that are only testing java.util classes.

### PR DESCRIPTION
**Pull request by jimfcarroll**
Fix test that doesn't compile when the JDK is Java 11 and above and remove a few tests that are only testing java.util classes.

The few tests removed seem that they are no longer testing custom collection classes and are now simply testing java.util collections.

### CODE REVIEW

File: common/common-lang/src/test/java/com/twelvemonkeys/util/CollectionAbstractTest.java [hunk 1]
```diff
@@ -1136,7 +1136,7 @@ public void testCollectionToArray2() {
         verifyAll();
 
         try {
-            array = collection.toArray(null);
+            array = collection.toArray((Object[])null);
```
Comment (haraldk) [inline common/common-lang/src/test/java/com/twelvemonkeys/util/CollectionAbstractTest.java]: This change looks ok. Please insert a space between the cast and the `null` to follow code style.

File: common/common-lang/src/test/java/com/twelvemonkeys/util/CollectionUtilTest.java [hunk 2]
```diff
@@ -228,34 +228,12 @@ public void testArrayListIteratorRange() {
         assertCorrectListIterator(CollectionUtil.iterator(new String[] {"foo", "bar", "baz", "boo"}, 1, 2), new String[] {"bar", "baz"});
     }
 
-    @Test
```
Comment (haraldk) [inline common/common-lang/src/test/java/com/twelvemonkeys/util/CollectionUtilTest.java]: Why did you remove these tests? I don't see anything wrong with these?

**Comment by haraldk** [comment]
I believe the problem you face is due to a new method in JDK 11, not JDK 8. But I guess it won't hurt making things forward compatible. :-)

File: common/common-lang/src/test/java/com/twelvemonkeys/util/CollectionUtilTest.java (hunk 2 above)
Comment (jimfcarroll) [reply common/common-lang/src/test/java/com/twelvemonkeys/util/CollectionUtilTest.java]: I could be wrong but it appears these tests don't test any of your code. The methods `assertCorrectListIterator` (which I made static just to be sure they didn't depend on the state of something set on the test class) have no dependencies on your custom collection classes and nothing passed into them in these test do either. It appears your testing `java.util.Arrays.asList`. 

Originally I thought all of your tests in this section did this but I see now you've created a test bed for you collection extentions. Did I miss something in this case?

File: common/common-lang/src/test/java/com/twelvemonkeys/util/CollectionAbstractTest.java (hunk 1 above)
Comment (jimfcarroll) [reply common/common-lang/src/test/java/com/twelvemonkeys/util/CollectionAbstractTest.java]: Okay. I added the space. I can squash the commits eventually if you'd prefer. Just let me know once I get it right. Thanks.

**Comment by haraldk** [comment]
Manually submitted the important parts of this PR, leaving the rest as-is.
